opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -218,13 +234,13 @@ impl crate::trace::Tracer for Tracer {
             self.process_sampling_result(sampling_result, parent_span_context)
         } else if no_parent || remote_parent {
```

Comment:
Maybe we can separate those two situations. If there is no parent, we build a parent context with the invalid span context. If there is a remote parent, we build a temporary span with the remote parent. Then we pass that context into `make_sampling_decision` or just use the sampler to make the sample decision.

Reply:
yeah this could have been a path too, was trying to do it without having to check before hand to know which one of these cases it was. But as this is all internal we should be able to keep iterating on this in the future without breaking changes